lambda-test/build.gradle.kts

```diff
@@ -0,0 +1,69 @@
+plugins {
+    id("java")
+}
+
+group = "org.example"
+version = "1.0-SNAPSHOT"
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation("org.junit.jupiter:junit-jupiter-api:5.8.1")
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.8.1")
+}
+
+tasks.getByName<Test>("test") {
+    useJUnitPlatform()
+}
+
+val printContents: (Project) -> Unit by rootProject.extra
```

Comment:
Okay... and if you put just `by extra` instead of `by rootProject.extra`... how would that differ?

Reply:
It will throw an error since this project's extra has no printContents value. That's another difference than in Groovy DSL. Example can be seen here: [developer.android](https://developer.android.com/studio/build/gradle-tips#configure-project-wide-properties)

Reply:
Also by changing:
val addElementNoProjectAllScope: (Unit) -> Unit by **extra**
to
val addElementNoProjectAllScope: (Unit) -> Unit by **rootProject.extra**

I can grab a different lambda that has captured the project variable from the rootProject and thus get a different behaviour💥

Reply:
:ups: yeah, I see now, indeed